mbuild/tests/test_lammpsdata.py

```diff
@@ -456,65 +493,173 @@ def test_lj_pairs(self, ethane):
-                    bonds.append(float(fi.readline().split()[1]))
-                    assert np.allclose(sorted(bonds), [49742.424, 63106.06])
+                    line = fi.readline().split()
+                    epsilon = float(line[1])
+                    sigma = float(line[2])
+                    assert np.isclose(epsilon, 0.066)
+                    assert np.isclose(sigma, 3.5)
                     checked_section = True
 
-    def test_lj_angles(self, ethane):
-        from foyer import Forcefield
+    def test_lj_bonds(self, ethane, lj_save):
+        fn = lj_save(ethane, Path.cwd())
+        checked_section = False
+        ethane_bondsk = (
+            np.array([224262.4, 284512.0]) * (0.35) ** 2 / (0.276144) / 2
+        )  # kj/mol/nm**2 to lammps
+        ethane_bondsreq = np.array([0.1529, 0.109]) * 0.35 ** -1  # unitless
+        ethane_lj_bonds = zip(ethane_bondsk, ethane_bondsreq)
+        with open(fn, "r") as fi:
+            while not checked_section:
+                line = fi.readline()
+                if "Bond Coeffs" in line:
+                    fi.readline()
+                    for bond_params in ethane_lj_bonds:
+                        print(bond_params)
+                        line = fi.readline()
+                        assert np.allclose(
+                            (float(line.split()[1]), float(line.split()[2])),
+                            bond_params,
+                            atol=1e-3,
+                        )
+                    checked_section = True
 
-        OPLSAA = Forcefield(name="oplsaa")
-        structure = OPLSAA.apply(ethane)
-        write_lammpsdata(
-            filename="lj.lammps", structure=structure, unit_style="lj"
-        )
+    def test_real_bonds(self, ethane, real_save):
+        fn = real_save(ethane, Path.cwd())
+        checked_section = False
+        ethane_bondsk = (
+            np.array([224262.4, 284512.0]) / KCAL_TO_KJ * ANG_TO_NM ** 2 / 2
+        )  # kj/mol/nm**2 to lammps
+        ethane_bondsreq = np.array([0.1529, 0.109]) / ANG_TO_NM
+        ethane_real_bonds = zip(ethane_bondsk, ethane_bondsreq)
+        with open(fn, "r") as fi:
+            while not checked_section:
+                line = fi.readline()
+                if "Bond Coeffs" in line:
+                    fi.readline()
+                    for bond_params in ethane_real_bonds:
+                        print(bond_params)
+                        line = fi.readline()
+                        assert np.allclose(
+                            (float(line.split()[1]), float(line.split()[2])),
+                            bond_params,
+                            atol=1e-3,
+                        )
+                    checked_section = True
 
+    def test_lj_angles(self, ethane, lj_save):
+        fn = lj_save(ethane, Path.cwd())
         checked_section = False
-        with open("lj.lammps", "r") as fi:
+        ethane_anglesk = (
+            np.array([313.8, 276.144]) / 0.276144 / 2
```

Comment:
Do we also need to account for k/2 for the angle force constants as well? I see the fix here but not in lammpsdata.py

Reply:
Same response as above.

Reply:
See comments above and below.